pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -153,10 +153,19 @@ private static Set<ValidationType> parseTypesToSkipString(@Nullable String types
    *   <li>Table name shouldn't contain dot or space in it</li>
    * </ul>
    */
-  public static void validateTableName(TableConfig tableConfig) {
+  public static void validateTableName(TableConfig tableConfig, boolean allowDots) {
     String tableName = tableConfig.getTableName();
-    if (tableName.contains(".") || tableName.contains(" ")) {
-      throw new IllegalStateException("Table name: '" + tableName + "' containing '.' or space is not allowed");
+    int dotCount = StringUtils.countMatches(tableName, '.');
+    // For transitioning into full [database_name].[table_name] support, we allow the table name
+    // with one dot at max, so the admin may create mydb.mytable with a feature knob.
+    if (allowDots && dotCount > 1) {
+      throw new IllegalStateException("Table name: '" + tableName + "' containing more than one '.' is not allowed");
+    }
+    if (!allowDots && dotCount > 0) {
+      throw new IllegalStateException("Table name: '" + tableName + "' containing '.' is not allowed");
+    }
+    if (tableName.contains(" ")) {
```

Comment:
Not introduced in this PR, but more robust
```suggestion
    if (StringUtils.containsWhitespace(tableName)) {
```